server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableDefinitionUtil.java

```diff
@@ -518,17 +521,22 @@ private VariableWithLevelNode getVariableForConditional(VariableDefinitionNode d
     Locality redefinesLocality = redefinesNameAndLocality.getLocality();
     VariableUsageNode redefineUsage = new VariableUsageNode(redefinesName, redefinesLocality);
     variableNode.addChild(redefineUsage);
-    VariableWithLevelNode variableForRedefine = getPrecedingVariableForRedefine(definitionNode, redefinesName);
-    if (variableForRedefine == null || !variableForRedefine.getName().equals(redefinesName)) {
+    List<VariableWithLevelNode> eligibleNodesForRedefine =
+        getEligibleNodesForRedefine(definitionNode);
+    Optional<VariableWithLevelNode> allowedRedefinedNode =
+        eligibleNodesForRedefine.stream()
+            .filter(node -> node.getName().equals(redefinesName))
+            .findFirst();
+    if (eligibleNodesForRedefine.isEmpty() || !allowedRedefinedNode.isPresent()) {
       errors.add(
           SyntaxError.syntaxError()
               .severity(SEVERITY)
               .messageTemplate(MessageTemplate.of(REDEFINE_IMMEDIATELY_FOLLOW, redefinesName))
               .locality(redefinesLocality)
               .build());
     } else {
-      variableForRedefine.addUsage(redefineUsage);
-      if (variableForRedefine.getLevel() != definitionNode.getLevel()) {
+      allowedRedefinedNode.get().addUsage(redefineUsage);
+      if (getEffectiveLevel(allowedRedefinedNode.get()) != getEffectiveLevel(definitionNode)) {
```

Comment:
Is it make sense to drop this check completely?

Reply:
Yes.

Reply:
Have kept this for level 77. level 77 can only be redefined by level 77.

Reply:
It looks overcomplicated.
As I understand it it can be like this: if one of the levels is 77 the other also must be 77. Otherwise, it doesn't matter what the effective level is.

Reply:
updated